lib/Auth/Process/UpdateAUP.php

```diff
@@ -63,7 +63,14 @@ public function process(&$state)
             return;
         }
         // Check if user is in blacklist
-        if(!empty($this->config['userIdAttribute']) && !empty($this->config['userIdBlacklist']) &&  !empty(array_intersect($state['Attributes'][$this->config['userIdAttribute']], $this->config['userIdBlacklist']))) {
+        // or $state['rciamAttributes']['aup'] is empty
+        if ((!empty($this->config['userIdAttribute']) && !empty($this->config['userIdBlacklist']) && !empty(
+            array_intersect(
+              $state['Attributes'][$this->config['userIdAttribute']],
+              $this->config['userIdBlacklist']
+            )
+            ))
+          || empty($state['rciamAttributes']['aup'])) {
```

Comment:
Add debug messages:
1. `"[aup:UpdateAUP] Skipping blacklisted user with id " . $state['Attributes'][$this->config['userIdAttribute']`
2. "[aup:UpdateAUP] No AUP information found in state - skipping"

Reply:
done